docs/slicer_cli_plugins.rst

```diff
@@ -0,0 +1,255 @@
+      <file fileExtensions=".anot" reference="inputImageFile">
+        <name>outputNucleiAnnotationFile</name>
+        <label>Output Nuclei Annotation File</label>
+        <description>Output nuclei annotation file (*.anot)</description>
+        <channel>output</channel>
+        <index>1</index>
+      </file>
+    </parameters>
+    <parameters advanced="true">
+      ...
+    </parameters>
+  </executable>
+
+Requirements for the XML File
+-----------------------------
+
+The xml file with the specification of the inputs and outputs of an algorithm is required.
+
+Many python algorithms use the ``CLIArgumentParser`` class to facilitate parsing the command line.  This can be imported via ``from ctk_cli import CLIArgumentParser``.  There are some ``slicer_cli_web`` specific extensions to the format; to use those, do ``from slicer_cli_web import CLIArgumentParser`` instead.  This has the virtue that the algorithm can parse the command line based on the xml.  This might look like:
+
+.. code-block:: python
+
+  if __name__ == '__main__':
+
+      args = CLIArgumentParser().parse_args()
+      # The parsed arguments can be accessed by the return value attributes
+      # value = args.parameter_name
+
+Alternatively, an algorithm can parse the command line itself, using the xml purely to inform programs that run the algorithm what parameters are required or available.
+
+Overall Section
++++++++++++++++
+
+There are several requirements for the xml.  At the minimum, it needs to contain a top level ``executable`` node containing a ``title`` and a ``description``.  This top level node will also always contain one or more ``parameter`` nodes (see below)i.  It can optionally contain ``license``, ``contributor``, ``acknowledgements`` (these three are used as part of the internal description), ``version``, (reported when ``--version`` is specified), ``category``, ``documentation-url``, ``index`` (these three are mostly ignored).  These optional nodes are expected to contain text.
+
+The minimal example is:
+
+.. code-block:: xml
+
+  <?xml version="1.0" encoding="UTF-8"?>
+  <executable>
+    <title>My Algorithm title</title>
+    <description>My algorithm description</description>
+    <parameters>
+      ...
+    </parameters>
+  </executable>
+
+Parameter Sections
+++++++++++++++++++
+
+Parameters consist of all of the inputs and outputs for the algorithm.  A parameter node must have ``label`` and ``description`` nodes.  It may contain an ``advanced`` attribute that is a binary value of ``true`` or ``false``.  User interfaces use this to determine whether the parameters should be exposed by default.  The parameter block contains any number of value nodes.
+
+The minimal example is:
+
+.. code-block:: xml
+
+  <parameters advanced="false">
+    <label>A block of parameters</label>
+    <description>A description of why these parameters are grouped together</description>
+    ...
+  </parameters>
+
+Value Sections
+++++++++++++++
```

Comment:
If I am understanding the output of the xml it looks as though the `Value Section` is a subset of the `Parameter Section`. If that is the case I don't think that is obvious from the description here. If that is not the case it looks as though this section is optional and should be noted here.